kaitai/struct_test.go

```diff
@@ -0,0 +1,46 @@
+package kaitai
+
+import (
+	"bytes"
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+// Declare two struct simulating types generated by Kaitai Struct compiler
+type oneStruct struct {
+	one int
+	_io *Stream
+}
+
+func (s *oneStruct) Kaitai_IO() *Stream {
+	return s._io
+}
+
+type twoStruct struct {
+	two int
+	_io *Stream
+}
+
+func (s *twoStruct) Kaitai_IO() *Stream {
+	return s._io
+}
+
+func WorkWithStruct(s Struct, t *testing.T, expectedSize int) {
+	actualSize, err := s.Kaitai_IO().Size()
+	assert.Equal(t, actualSize, int64(expectedSize))
+	assert.Nil(t, err)
+}
+
+func TestKaitaiStruct(t *testing.T) {
+	// Instantiate streams for the structs
+	oneStream := NewStream(bytes.NewReader([]byte("a")))
+	twoStream := NewStream(bytes.NewReader([]byte("ab")))
+
+	// Instantiate the structs
+	one := oneStruct{1, oneStream}
+	two := twoStruct{2, twoStream}
+
+	// Check if the structs implement the Struct interface
+	WorkWithStruct(&one, t, 1)
+	WorkWithStruct(&two, t, 2)
```

Comment:
It's almost misleading that both `one.one == one._io.size() == 1` and `two.two == two._io.size() == 2`, even if `one.one`/`two.two` are totally unrelated to `one._io.size()`/`two._io.size()` :)

I think making it clear that they are unrelated by using unique numbers everywhere would be less mentally challenging. For example, change `1` and `2` here to `42` and `37`, cuz why not:

```go
	one := oneStruct{42, oneStream}
	two := twoStruct{37, twoStream}
```

... or, obviously, we can change the lengths of the streams so that they are not `1`, `2` respectively.

Reply:
Done both, take a look?